src/server/routes/entity/entity.js

```diff
@@ -274,6 +273,135 @@ export function addNoteToRevision(req: PassportRequest, res: $Response) {
+	orm, transacting, isNew: boolean, newRevision: any, mainEntity: any,
+	updatedEntities: [], editorID: number, note: string
+) {
+	const parentRevisionIDs = _.compact(_.uniq(updatedEntities.map(
+		(entityModel) => entityModel.get('revisionId')
+	)));
+
+	const entitiesSavedPromise = Promise.all(
+		_.map(updatedEntities, (entityModel) => {
+			entityModel.set('revisionId', newRevision.get('id'));
+
+			const shouldInsert =
+				entityModel.get('bbid') === mainEntity.get('bbid') && isNew;
+			const method = shouldInsert ? 'insert' : 'update';
+			return entityModel.save(null, {method, transacting});
+		})
+	);
+
+	const editorUpdatePromise =
+		utils.incrementEditorEditCountById(orm, editorID, transacting);
+
+	const notePromise = _createNote(
+		orm, note, editorID, newRevision, transacting
+	);
+
+	const parentsAddedPromise =
+		setParentRevisions(transacting, newRevision, parentRevisionIDs);
+
+	/** model.save returns a refreshed model */
+	const [savedEntities, ...others] = await Promise.all([
+		entitiesSavedPromise,
+		editorUpdatePromise,
+		parentsAddedPromise,
+		notePromise
+	]);
+
+	return savedEntities.find(entityModel => entityModel.get('bbid') === mainEntity.get('bbid')) || mainEntity;
+}
+
+async function deleteRelationships(orm, transacting, mainEntity) {
+	const mainBBID = mainEntity.bbid;
+	const {RelationshipSet} = orm;
+	const {relationshipSet} = mainEntity;
+	const otherBBIDs = [];
+	const otherEntities = [];
+
+	if (relationshipSet) {
+		// Create a list of BBID's that is related to the deleted entity
+		relationshipSet.relationships.foreach((relationship) => {
+			if (relationship.sourceBbid === mainBBID) {
+				otherBBIDs.push(relationship.targetBbid);
+			}
+			else if (relationship.targetBbid === mainBBID) {
+				otherBBIDs.push(relationship.sourceBbid);
+			}
+		});
+
+		// Loop over the BBID's of other entites related to deleted entity
+		if (otherBBIDs.length) {
+			if (otherBBIDs.length == 0) {
+				return null;
+			}
+			await otherBBIDs.map(async (entityBbid) => {
+				const otherEntity = await getEntityByBBID(orm, transacting, entityBbid);
+
+				otherEntities.push(otherEntity);
+
+				const otherEntityRelationshipSet = await otherEntity.relationshipSet()
+					.fetch({require: false, transacting, withRelated: 'relationships'});
+
+				// Fetch other entity relationships to remove relation with the deleted entity
+				let otherEntityRelationships = otherEntityRelationshipSet.related('relationships').toJSON();
+
+				// Filter out entites related to deleted entity
+				otherEntityRelationships = otherEntityRelationships.filter(({sourceBbid, targetBbid}) =>
+					mainBBID !== sourceBbid && mainBBID !== targetBbid);
+
+				
+
+				const newRelationshipSet = await orm.func.relationship.updateRelationshipSets(
+					orm, transacting, otherEntityRelationshipSet, otherEntityRelationships
+				);
+
+				otherEntity.set(
+					'relationshipSetId',
+					newRelationshipSet.get('id')
```

Comment:
I was wrong about how this works: `updateRelationshipSets` returns an object with the involved entities' BBIDs as keys and the RelationshipSets as value.

So you'll need to do `newRelationshipSet[entityBbid].get('id')` instead to target the correct RelationshipSet